src/ServiceControl.Plugin.Nsb5.Heartbeat/Heartbeats.cs

```diff
@@ -49,7 +49,11 @@ public void Dispose()
         {
             if (heartbeatTimer != null)
             {
-                heartbeatTimer.Dispose();
+                using (var manualResetEvent = new ManualResetEvent(false))
+                {
+                    heartbeatTimer.Dispose(manualResetEvent);
+                    manualResetEvent.WaitOne();
```

Comment:
would a timeout be sensible here https://msdn.microsoft.com/en-us/library/cc189907(v=vs.110).aspx  ?